### PR TITLE
Added an info reminding user that if no time_budget and no max_iter is specified, then effectively zero-shot AutoML is used

### LINKED ISSUE #798: Larger dimensionality sparse array inputs cause many FLAML use cases to fail
Classification and regression models often leverage sparse arrays. For example when composing a Sklearn pipeline, sparse arrays are formed as the output of categorical OneHot transforms and ngramming like CountVectorizer and TfidfVectorizer.

These sparse arrays are passed into the learners (eg LightGbm, Xgboost, LogisticRegression), and these learners are designed to support these sparse arrays efficiently allowing large dimensionality classifiers to be trained.

The fix I think it to ensure we are keeping arrays in their sparse array formats, eg `scipy.sparse.csr_matrix` at least when passing into the underlying learner and preferably through the handling of the data by FLAML as well.

Here are a few repro examples that illustrate the issue, note that generally any example using OneHot, CountVectorizer, or TfidfVectorizer will repro similarly and these transforms are common:

Example 1 - LightGbm regression example repro. The FLAML will execution part looks to densify the sparse matrix output from the ColumnTransformer (takes up 5GB RAM), and gets stuck failing to fit the learner. A couple notes:
* This is just 700K distinct categorical features, but the same problem is present for smaller numbers like ~70K distinct features too. LightGbm can handle this.
* Without FLAML it takes 3 seconds to fit.
* With FLAML it hangs.
* Python takes up 5GB RAM as soon as it tries to start the FLAML part.
```python
import numpy as np
import pandas as pd

from sklearn.preprocessing import OneHotEncoder
from sklearn.pipeline import Pipeline
from sklearn.compose import ColumnTransformer
from sklearn.metrics import accuracy_score

from flaml import AutoML

from lightgbm import LGBMRegressor

# Create an example dataframe with random integers and a target column
df = pd.DataFrame(np.random.randint(0, 1000000, size=(500000, 2)), columns=['A', 'B'])
df['target'] = ((df['A'] + df['B']) > 1000000).astype(int)
print(df.head(10))

# Print dataset statistics
print("Dataset statistics:")
print(f"Number of rows: {len(df)}")
print(f"Number of unique categorical features: {df['A'].nunique() + df['B'].nunique()}")

# Build a sklearn pipeline that one-hot encodes the integers

# Create the pipeline with a LightGBM regression model
pipeline = Pipeline([
    ('preprocessor', ColumnTransformer([
        ('onehot', OneHotEncoder(handle_unknown='ignore'), ['A', 'B'])
    ])),
    ('classifier', LGBMRegressor())
])

# Fit the model while timing the process
import time
start = time.time()
pipeline.fit(df[['A', 'B']], df['target'])
end = time.time()

# Print the metrics
print("LightGBM")
print("MAE:", np.mean(np.abs(pipeline.predict(df[['A', 'B']]) - df['target'])))
print("Number of parameters:", pipeline['classifier'].n_features_)
print("Fitting took {} seconds".format(end - start))

# Now fit a FLAML model instead with LightGBM and the same data
pipeline = Pipeline(
    [
        (
            'preprocessor', ColumnTransformer(
                [
                    ('onehot', OneHotEncoder(handle_unknown='ignore'), ['A', 'B'])
                ]
            )
        ),
        (
            'classifier', AutoML(
                task='regression',
                estimator_list=['lgbm'],
            )
        )
    ]
)
start = time.time()
pipeline.fit(df[['A', 'B']], df['target'])
end = time.time()

# Print the metrics
print("FLAML")
print("MAE:", np.mean(np.abs(pipeline.predict(df[['A', 'B']]) - df['target'])))
print("Fitting took {} seconds".format(end - start))
```

Output:
```
        A       B  target
0  392214  547064       0
1  733104   70527       0
2  435315  879075       1
3    3221  115858       0
4  507700  113589       0
5  746200  312285       1
6  671553  164952       0
7  486703  540223       1
8  540288  793887       1
9  201758  563042       0
Dataset statistics:
Number of rows: 500000
Number of unique categorical features: 786525
LightGBM
MAE: 0.499999640448
Number of parameters: 786525
Fitting took 3.1742513179779053 seconds
[flaml.automl: 11-04 21:18:01] {2600} INFO - task = regression
[flaml.automl: 11-04 21:18:01] {2602} INFO - Data split method: uniform
[flaml.automl: 11-04 21:18:01] {2605} INFO - Evaluation method: holdout
[flaml.automl: 11-04 21:18:01] {2727} INFO - Minimizing error metric: 1-r2
[flaml.automl: 11-04 21:18:01] {2777} WARNING - No search budget is provided via time_budget or max_iter. Training only one model per estimator. To tune hyperparameters for each estimator, please provide budget either via time_budget or max_iter.
[flaml.automl: 11-04 21:18:01] {2869} INFO - List of ML learners in AutoML Run: ['lgbm']
```

Example 2 - FLAML binary classification without learner specified (note: binary classification limited to lgbm works)
* Without FLAML it takes 3.2 seconds to fit
* With FLAML it hangs, but does NOT take up a large amount of RAM right away. Maybe an iteration issue rather than a dense conversion issue at least at first?

```python
import numpy as np
import pandas as pd

from sklearn.preprocessing import OneHotEncoder
from sklearn.pipeline import Pipeline
from sklearn.compose import ColumnTransformer
from sklearn.metrics import accuracy_score

from flaml import AutoML

from lightgbm import LGBMRegressor, LGBMClassifier

# Create an example dataframe with random integers and a target column
df = pd.DataFrame(np.random.randint(0, 1000000, size=(500000, 2)), columns=['A', 'B'])
df['target'] = ((df['A'] + df['B']) > 1000000).astype(int)
print(df.head(10))

# Print dataset statistics
print("Dataset statistics:")
print(f"Number of rows: {len(df)}")
print(f"Number of unique categorical features: {df['A'].nunique() + df['B'].nunique()}")

# Build a sklearn pipeline that one-hot encodes the integers

# Create the pipeline with a LightGBM regression model
pipeline = Pipeline([
    ('preprocessor', ColumnTransformer([
        ('onehot', OneHotEncoder(handle_unknown='ignore'), ['A', 'B'])
    ])),
    ('classifier', LGBMClassifier())
])

# Fit the model while timing the process
import time
start = time.time()
pipeline.fit(df[['A', 'B']], df['target'])
end = time.time()

# Print the metrics
print("LightGBM")
print("MAE:", np.mean(np.abs(pipeline.predict(df[['A', 'B']]) - df['target'])))
print("Number of parameters:", pipeline['classifier'].n_features_)
print("Fitting took {} seconds".format(end - start))

# Now fit a FLAML model instead with LightGBM and the same data
pipeline = Pipeline(
    [
        (
            'preprocessor', ColumnTransformer(
                [
                    ('onehot', OneHotEncoder(handle_unknown='ignore'), ['A', 'B'])
                ]
            )
        ),
        (
            'classifier', AutoML(
                task='classification',
            )
        )
    ]
)
start = time.time()
pipeline.fit(df[['A', 'B']], df['target'])
end = time.time()

# Print the metrics
print("FLAML")
print("MAE:", np.mean(np.abs(pipeline.predict(df[['A', 'B']]) - df['target'])))
print("Fitting took {} seconds".format(end - start))
```

Output:
```
    A       B  target
0  435201   20896       0
1  650113  830984       1
2  296903  335422       0
3  733399  881420       1
4  968169  600231       1
5  243170   94188       0
6  303474  476584       0
7  950125  645745       1
8  607060  432579       1
9  741146  550150       1
Dataset statistics:
Number of rows: 500000
Number of unique categorical features: 787371
LightGBM
MAE: 0.499826
Number of parameters: 787371
Fitting took 3.257169246673584 seconds
[flaml.automl: 11-04 21:20:45] {2600} INFO - task = classification
[flaml.automl: 11-04 21:20:45] {2602} INFO - Data split method: stratified
[flaml.automl: 11-04 21:20:45] {2605} INFO - Evaluation method: holdout
[flaml.automl: 11-04 21:20:46] {2727} INFO - Minimizing error metric: 1-roc_auc
[flaml.automl: 11-04 21:20:46] {2777} WARNING - No search budget is provided via time_budget or max_iter. Training only one model per estimator. To tune hyperparameters for each estimator, please provide budget either via time_budget or max_iter.
[flaml.automl: 11-04 21:20:46] {2869} INFO - List of ML learners in AutoML Run: ['extra_tree', 'lgbm', 'rf', 'xgboost', 'xgb_limitdepth', 'lrl1']
[flaml.automl: 11-04 21:20:46] {3164} INFO - iteration 0, current learner extra_tree
```

### PR DESCRIPTION
**Pull request by jingdong00**
## Why are these changes needed?
When neither time_budget nor max_iter is specified, zero-shot ML is used. But the user can be unaware of this as no information may be provided. 

Added an info through logger that says 'Neither time_budegt nor max_iter is specified, zero-shot ML is used. '. 

## Related issue number

Closes #798 
## Checks

- [ x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ NA] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [NA ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x ] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by sonichi — COMMENTED** [review]
We already issue a warning in line 2821. Move the message to there?

**Comment by jingdong00** [comment]
> We already issue a warning in line 2821. Move the message to there?

Right. I'll move this later.

File: flaml/automl/automl.py [hunk 1]
```diff
@@ -2818,6 +2819,7 @@ def is_to_reverse_metric(metric, task):
                 " Training only one model per estimator."
                 " To tune hyperparameters for each estimator,"
                 " please provide budget either via time_budget or max_iter."
+                " Note that if both are not specified, zero-shot ML is used"
```
Comment (sonichi) [inline flaml/automl/automl.py:2822]:
```suggestion
                " Training only one model per estimator."
                " Zero-shot ML is used for certain tasks and estimators."                
                " To tune hyperparameters for each estimator,"
                " please provide budget either via time_budget or max_iter."
```

File: flaml/automl/automl.py [hunk 2]
```diff
@@ -2817,9 +2817,9 @@ def is_to_reverse_metric(metric, task):
             logger.warning(
                 "No search budget is provided via time_budget or max_iter."
                 " Training only one model per estimator."
+                " Zero-shot ML is used for certain tasks and estimators."
```
Comment (sonichi) [inline flaml/automl/automl.py:2820]:
```suggestion
                " Zero-shot AutoML is used for certain tasks and estimators."                
```

**Review by sonichi — APPROVED** [review]

**Pull request merged**